### PR TITLE
:bug: Source MixPanel: allow pulling data from different regions

### LINKED ISSUE #5880: MixPanel Connection
## Enviroment
- **Airbyte version**:  0.29.15-alpha
- **OS Version / Instance**:  	Ubuntu 16.04.7 LTS // GCP e2-standard-2 (2 vCPUs, 8 GB memory)
- **Deployment**:  Docker-compose
- **Source Connector and version**:  Mixpanel 0.1.0
- **Destination Connector and version**: Bigquery 0.1.1
- **Severity**: -
- **Step where error happened**:  Sync Job

## Current Behavior
Error when sync to MixPanel, after doing the checking of the Source and destination, it fails with Failed to fetch schema error

## Expected Behavior
Data should be synced to BigQuery

## Logs

<details>
<summary>LOG</summary> 

```

2021-09-07 13:30:07 INFO () TemporalAttemptExecution(get):111 - Executing worker wrapper. Airbyte version: 0.29.15-alpha
2021-09-07 13:30:07 INFO () LogClientSingleton(setJobMdc):146 - Setting docker job mdc
2021-09-07 13:30:07 INFO () LineGobbler(voidCall):85 - Checking if airbyte/source-mixpanel:0.1.0 exists...
2021-09-07 13:30:07 INFO () LineGobbler(voidCall):85 - airbyte/source-mixpanel:0.1.0 was found locally.
2021-09-07 13:30:07 INFO () DockerProcessFactory(create):146 - Preparing command: docker run --rm --init -i -v airbyte_workspace:/data -v /tmp/airbyte_local:/local -w /data/311d551a-9741-49c4-9155-d6a686ae56ad/0 --network host --log-driver none airbyte/source-mixpanel:0.1.0 discover --config source_config.json
2021-09-07 13:30:09 INFO () DefaultAirbyteStreamFactory(internalLog):110 - Using start_date: 2021-11-06, end_date: 2021-09-07
2021-09-07 13:30:09 ERROR () DefaultAirbyteStreamFactory(internalLog):108 - Stream engage_schema: 400 Bad Request - {"request": "/api/2.0/engage/properties", "error": "engage-distributed-query server error: 400 \nfor params: {'project_id': 2194296, 'limit': 2047, 'type': 'properties', 'properties_query_method': 'top_properties', 'query_id': '9dde25178fbd490181303bcef839264a', 'timezone': 'US/Pacific', 'enable_group_derived_files': True, 'enable_engage_query_derived_files': False, 'enable_arb_query_derived_files': False, 'enable_id_mapping': True, 'enable_async_shuffle': True}"}
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 - Traceback (most recent call last):
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -   File "/airbyte/integration_code/main.py", line 33, in <module>
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -     launch(source, sys.argv[1:])
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -   File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/entrypoint.py", line 117, in launch
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -     for message in source_entrypoint.run(parsed_args):
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -   File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/entrypoint.py", line 102, in run
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -     catalog = self.source.discover(logger, config)
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -   File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 78, in discover
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -     streams = [stream.as_airbyte_stream() for stream in self.streams(config=config)]
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -   File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/abstract_source.py", line 78, in <listcomp>
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -     streams = [stream.as_airbyte_stream() for stream in self.streams(config=config)]
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -   File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/streams/core.py", line 80, in as_airbyte_stream
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -     stream = AirbyteStream(name=self.name, json_schema=dict(self.get_json_schema()), supported_sync_modes=[SyncMode.full_refresh])
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -   File "/airbyte/integration_code/source_mixpanel/source.py", line 527, in get_json_schema
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -     for property_entry in schema_properties:
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -   File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/streams/http/http.py", line 240, in read_records
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -     response = self._send_request(request, request_kwargs)
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -   File "/airbyte/integration_code/source_mixpanel/source.py", line 98, in _send_request
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -     raise e
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -   File "/airbyte/integration_code/source_mixpanel/source.py", line 93, in _send_request
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -     return super()._send_request(request, request_kwargs)
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -   File "/usr/local/lib/python3.7/site-packages/backoff/_sync.py", line 94, in retry
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -     ret = target(*args, **kwargs)
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -   File "/usr/local/lib/python3.7/site-packages/backoff/_sync.py", line 94, in retry
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -     ret = target(*args, **kwargs)
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -   File "/usr/local/lib/python3.7/site-packages/airbyte_cdk/sources/streams/http/http.py", line 216, in _send_request
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -     response.raise_for_status()
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -   File "/usr/local/lib/python3.7/site-packages/requests/models.py", line 953, in raise_for_status
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 -     raise HTTPError(http_error_msg, response=self)
2021-09-07 13:30:09 ERROR () LineGobbler(voidCall):85 - requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://mixpanel.com/api/2.0/engage/properties
2021-09-07 13:30:09 INFO () TemporalAttemptExecution(lambda$getWorkerThread$2):155 - Completing future exceptionally...
io.airbyte.workers.WorkerException: Discover job subprocess finished with exit code 1
	at io.airbyte.workers.DefaultDiscoverCatalogWorker.run(DefaultDiscoverCatalogWorker.java:90) ~[io.airbyte-airbyte-workers-0.29.15-alpha.jar:?]
	at io.airbyte.workers.DefaultDiscoverCatalogWorker.run(DefaultDiscoverCatalogWorker.java:44) ~[io.airbyte-airbyte-workers-0.29.15-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:152) ~[io.airbyte-airbyte-workers-0.29.15-alpha.jar:?]
	at java.lang.Thread.run(Thread.java:832) [?:?]
2021-09-07 13:30:09 INFO () TemporalAttemptExecution(get):139 - Stopping cancellation check scheduling...
2021-09-07 13:30:09 WARN () POJOActivityTaskHandler$POJOActivityImplementation(execute):243 - Activity failure. ActivityId=a481db20-95e9-3378-a2a7-37e6387e63d1, activityType=Run, attempt=1
java.util.concurrent.ExecutionException: io.airbyte.workers.WorkerException: Discover job subprocess finished with exit code 1
	at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:395) ~[?:?]
	at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2063) ~[?:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.get(TemporalAttemptExecution.java:137) ~[io.airbyte-airbyte-workers-0.29.15-alpha.jar:?]
	at io.airbyte.workers.temporal.DiscoverCatalogWorkflow$DiscoverCatalogActivityImpl.run(DiscoverCatalogWorkflow.java:107) ~[io.airbyte-airbyte-workers-0.29.15-alpha.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:564) ~[?:?]
	at io.temporal.internal.sync.POJOActivityTaskHandler$POJOActivityInboundCallsInterceptor.execute(POJOActivityTaskHandler.java:277) ~[temporal-sdk-1.0.4.jar:?]
	at io.temporal.internal.sync.POJOActivityTaskHandler$POJOActivityImplementation.execute(POJOActivityTaskHandler.java:216) ~[temporal-sdk-1.0.4.jar:?]
	at io.temporal.internal.sync.POJOActivityTaskHandler.handle(POJOActivityTaskHandler.java:181) ~[temporal-sdk-1.0.4.jar:?]
	at io.temporal.internal.worker.ActivityWorker$TaskHandlerImpl.handle(ActivityWorker.java:192) ~[temporal-sdk-1.0.4.jar:?]
	at io.temporal.internal.worker.ActivityWorker$TaskHandlerImpl.handle(ActivityWorker.java:154) ~[temporal-sdk-1.0.4.jar:?]
	at io.temporal.internal.worker.PollTaskExecutor.lambda$process$0(PollTaskExecutor.java:73) ~[temporal-sdk-1.0.4.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
	at java.lang.Thread.run(Thread.java:832) [?:?]
Caused by: io.airbyte.workers.WorkerException: Discover job subprocess finished with exit code 1
	at io.airbyte.workers.DefaultDiscoverCatalogWorker.run(DefaultDiscoverCatalogWorker.java:90) ~[io.airbyte-airbyte-workers-0.29.15-alpha.jar:?]
	at io.airbyte.workers.DefaultDiscoverCatalogWorker.run(DefaultDiscoverCatalogWorker.java:44) ~[io.airbyte-airbyte-workers-0.29.15-alpha.jar:?]
	at io.airbyte.workers.temporal.TemporalAttemptExecution.lambda$getWorkerThread$2(TemporalAttemptExecution.java:152) ~[io.airbyte-airbyte-workers-0.29.15-alpha.jar:?]
	... 1 more

```

</details>

## Steps to Reproduce
1. Create MixPanel Source configuration
2. Create BigQuery Destination configuration
3. Link them and try to sync

**NOTE** After reading a bit, I found that some 400 errors are thrown because I'm using the "eu" mixpanel version, and european resident zone are using another host. Maybe its just a matter to be able to choose which source are you needed like in bigquery,  (info here: https://developer.mixpanel.com/reference/overview)

### PR DESCRIPTION
**Pull request by yaroslav-dudar**
## What
Added project region option #5880

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## Pre-merge Checklist
Expand the relevant checklist and delete the others. 

<details><summary> <strong> New Connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>
</details>

<details><summary> <strong> Updating a connector </strong></summary>
<p>
   
#### Community member or Airbyter
   
- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector version bumped like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</p>
</details>

<details><summary> <strong> Connector Generator </strong> </summary>
<p>
   
- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed.
</p>
</details>

### CODE REVIEW

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/spec.json [hunk 1]
```diff
@@ -38,6 +38,39 @@
         "description": "The default value to use if no bookmark exists for an endpoint. Default is 1 year ago.",
         "examples": ["2021-11-16"],
         "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}(T[0-9]{2}:[0-9]{2}:[0-9]{2}Z)?$"
+      },
+      "region": {
```
Comment (midavadim) [inline airbyte-integrations/connectors/source-mixpanel/source_mixpanel/spec.json:42]: I think we should do this update in a such way so existing users do not need to modify their config files and add region param.  

If region param is missing then we should use US by default.

**Review by midavadim — CHANGES_REQUESTED** [review]

**Review by midavadim — APPROVED** [review]

File: docs/integrations/sources/mixpanel.md [hunk 2]
```diff
@@ -43,14 +43,18 @@ The Mixpanel connector should not run into Mixpanel API limitations under normal
 
 * Mixpanel API Secret
 
+* Project region `US` or `EU`
+
 ### Setup guide
 
 Please read [Find API Secret](https://help.mixpanel.com/hc/en-us/articles/115004502806-Find-Project-Token-).
 
+If you get `400` response codes, make sure that project region matches mixpanel account region. See detail [here](https://help.mixpanel.com/hc/en-us/articles/360039135652-Data-Residency-in-EU)
```
Comment (sherifnada) [inline docs/integrations/sources/mixpanel.md:52]:
```suggestion
Select the correct region (EU or US) for your Mixpanel project. See detail [here](https://help.mixpanel.com/hc/en-us/articles/360039135652-Data-Residency-in-EU)
```

File: airbyte-integrations/connectors/source-mixpanel/source_mixpanel/spec.json [hunk 3]
```diff
@@ -38,6 +38,39 @@
...
+        "title": "Project region",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-mixpanel/source_mixpanel/spec.json:43]: You can do this more simply by encoding it as a string e.g: 

```
"region: {
  "type": "string", 
  "enum": ["US", "EU"]
}
```

**Review by sherifnada — APPROVED** [review]

**Review by sherifnada — APPROVED** [review]

**Pull request merged**